Source/Scene/ModelExperimental/SkinningPipelineStage.js

```diff
@@ -0,0 +1,118 @@
+import combine from "../../Core/combine.js";
+import ShaderDestination from "../../Renderer/ShaderDestination.js";
+import SkinningStageVS from "../../Shaders/ModelExperimental/SkinningStageVS.js";
+import VertexAttributeSemantic from "../VertexAttributeSemantic.js";
+
+/**
+ * The skinning pipeline stage processes the joint matrices of a skinned primitive.
+ *
+ * @namespace SkinningPipelineStage
+ *
+ * @private
+ */
+
+const SkinningPipelineStage = {};
+SkinningPipelineStage.name = "SkinningPipelineStage"; // Helps with debugging
+
+SkinningPipelineStage.FUNCTION_ID_GET_SKINNING_MATRIX = "getSkinningMatrix";
+SkinningPipelineStage.FUNCTION_SIGNATURE_GET_SKINNING_MATRIX =
+  "mat4 getSkinningMatrix()";
+
+/**
+ * This pipeline stage processes the joint matrices of a skinned primitive, adding
+ * the relevant functions and uniforms to the shaders. The joint and weight attributes
+ * themselves are processed in the geometry pipeline stage.
+ *
+ * Processes a primitive. This stage modifies the following parts of the render resources:
+ * <ul>
+ *  <li> adds the uniform declaration for the joint matrices in the vertex shader
+ *  <li> adds the function to compute the skinning matrix in the vertex shader
+ * </ul>
+ *
+ * @param {PrimitiveRenderResources} renderResources The render resources for this primitive.
+ * @param {ModelComponents.Primitive} primitive The primitive.
+ * @private
+ */
+
+SkinningPipelineStage.process = function (renderResources, primitive) {
+  const shaderBuilder = renderResources.shaderBuilder;
+
+  shaderBuilder.addDefine("HAS_SKINNING", undefined, ShaderDestination.VERTEX);
+  addGetSkinningMatrixFunction(shaderBuilder, primitive);
+
+  const runtimeNode = renderResources.runtimeNode;
+  const jointMatrices = runtimeNode.computedJointMatrices;
+
+  shaderBuilder.addUniform(
+    "mat4",
+    `u_jointMatrices[${jointMatrices.length}]`,
+    ShaderDestination.VERTEX
+  );
+
+  shaderBuilder.addVertexLines([SkinningStageVS]);
+
+  const uniformMap = {
+    u_jointMatrices: function () {
+      return jointMatrices;
```

Comment:
I doubt this would change at runtime, but nevertheless it would be good practice to use `runtimeNode.jointMatrices` to be more future-proof. This way the latest value is always used, not a constant.